contracts/tests/MockYearnToken.vy

```diff
@@ -0,0 +1,1681 @@
+    self._assessFees(msg.sender, gain)
+
+    # Returns are always "realized gains"
+    self.strategies[msg.sender].totalGain += gain
+
+    # Outstanding debt the Strategy wants to take back from the Vault (if any)
+    # NOTE: debtOutstanding <= StrategyParams.totalDebt
+    debt: uint256 = self._debtOutstanding(msg.sender)
+    debtPayment: uint256 = min(_debtPayment, debt)
+
+    if debtPayment > 0:
+        self.strategies[msg.sender].totalDebt -= debtPayment
+        self.totalDebt -= debtPayment
+        debt -= debtPayment
+        # NOTE: `debt` is being tracked for later
+
+    # Compute the line of credit the Vault is able to offer the Strategy (if any)
+    credit: uint256 = self._creditAvailable(msg.sender)
+
+    # Update the actual debt based on the full credit we are extending to the Strategy
+    # or the returns if we are taking funds back
+    # NOTE: credit + self.strategies[msg.sender].totalDebt is always < self.debtLimit
+    # NOTE: At least one of `credit` or `debt` is always 0 (both can be 0)
+    if credit > 0:
+        self.strategies[msg.sender].totalDebt += credit
+        self.totalDebt += credit
+
+    # Give/take balance to Strategy, based on the difference between the reported gains
+    # (if any), the debt payment (if any), the credit increase we are offering (if any),
+    # and the debt needed to be paid off (if any)
+    # NOTE: This is just used to adjust the balance of tokens between the Strategy and
+    #       the Vault based on the Strategy's debt limit (as well as the Vault's).
+    totalAvail: uint256 = gain + debtPayment
+    if totalAvail < credit:  # credit surplus, give to Strategy
+        self.erc20_safe_transfer(self.token.address, msg.sender, credit - totalAvail)
+    elif totalAvail > credit:  # credit deficit, take from Strategy
+        self.erc20_safe_transferFrom(self.token.address, msg.sender, self, totalAvail - credit)
+    # else, don't do anything because it is balanced
+
+    # Update reporting time
+    self.strategies[msg.sender].lastReport = block.timestamp
+    self.lastReport = block.timestamp
+    self.lockedProfit = gain # profit is locked and gradually released per block
+
+    log StrategyReported(
+        msg.sender,
+        gain,
+        loss,
+        debtPayment,
+        self.strategies[msg.sender].totalGain,
+        self.strategies[msg.sender].totalLoss,
+        self.strategies[msg.sender].totalDebt,
+        credit,
+        self.strategies[msg.sender].debtRatio,
+    )
+
+    if self.strategies[msg.sender].debtRatio == 0 or self.emergencyShutdown:
+        # Take every last penny the Strategy has (Emergency Exit/revokeStrategy)
+        # NOTE: This is different than `debt` in order to extract *all* of the returns
+        return Strategy(msg.sender).estimatedTotalAssets()
+    else:
+        # Otherwise, just return what we have as debt outstanding
+        return debt
+
+
+@external
```

Comment:
can remove this file but kept it for reference for where bytecode in deployMockYVault.js came from